src/main/java/org/kohsuke/github/GHPullRequest.java

```diff
@@ -200,7 +200,7 @@ public String getMergeCommitSha() throws IOException {
      * Depending on the original API call where this object is created, it may not contain everything.
      */
     private void populate() throws IOException {
-        if (merged_by!=null)    return; // already populated
+        if (mergeable_state != null)    return; // already populated by id
```

Comment:
merged_by may be null if the PR was unmerged, leading to an API call every time we call populate()